parsers/Parser.js

```diff
@@ -7,31 +9,97 @@ module.exports = class Parser {
         this.document = document;
     }
 
-    parse() {
-        return [];
+    getRules() {
+        const rules = this.rules;
+
+        return rules && rules.length ? this.flagDuplicates(rules) : [];
     }
 
-    getRules() {
-        let rules = this.parse();
+    get rules() {
+        // This default implementation fits for for YAML and JSON parsers.
+        // JS & Package parsers override it.
```

Comment:
I would prefer that this class not know about extensions to it.  Could these comments be removed?